src/mapping/Mapping.cpp

```diff
@@ -86,10 +114,20 @@ bool Mapping::requiresGradientData() const
   return _requiresGradientData;
 }
 
+void Mapping::map(int inputDataID, int outputDataID, Eigen::VectorXd &initialGuess)
+{
+  PRECICE_ASSERT(_initialGuess == nullptr);
+  _initialGuess = &initialGuess;
+  map(inputDataID, outputDataID);
+  _initialGuess = nullptr;
+}
```

Comment:
We still need to remove this one before merging, right?

Reply:
I would leave it for now and port the mappings bit by bit. There are many changes to be made.

Reply:
Really? I thought you ported pretty much every mapping here already? Maybe I misunderstood some missing complexity. Then please open a follow-up issue to not forget.

Reply:
The Mapping hierarchy isn't the problem.

The sampling simplifies tests dramatically as mapping tests only need to define meshes and samples now. Mesh data isn't necessary anymore.

So while it simplifies many tests, it also requires to rewrite big parts of many tests.

Reply:
:+1:
-> (good first) issue

Reply:
https://github.com/precice/precice/issues/1672